addons/website_event_track/static/src/js/website_event_track_proposal.js

```diff
@@ -0,0 +1,182 @@
+odoo.define('website_event_track.website_event_track_proposal', function (require) {
+    'use strict';
+
+    var core = require('web.core');
+    var publicWidget = require('web.public.widget');
+    var utils=require('web.utils');
+
+    var _t = core._t;
+
+    publicWidget.registry.websiteEventProposalForm = publicWidget.Widget.extend({
+
+    selector: '.o_website_event_track_proposal_form',
+    events: {
+        'input input[name="partner_name"]': '_onNameInputPropagateOnContact',
+        'click .o_wetrack_add_contact_information': '_onToggleContactInformationClick',
+        'click .o_wetrack_proposal_submit_button': '_send',
+    },
+
+    /**
+     * Check validity of input fields : empty required fields or bad formatted ones are 
+     * marked as is-invalid and o_has_error.
+     * 
+     * @private
+     */
+    _check_error_fields: function () {
+        var form_valid = true;
+
+        this.$('.form-group').each(function (k, field) { 
```

Comment:
Also naming remark: what is "k" ? shoudn't field be named input or something like that ?

Reply:
Yes, done, I was confused by the different syntaxes here.